api/src/main/java/org/datacleaner/configuration/DataCleanerEnvironment.java

```diff
@@ -62,4 +62,9 @@
      */
     public InjectionManagerFactory getInjectionManagerFactory();
 
+    /**
+     * @see DescriptorProviderStateNotifier
+     * @return instance of notifier
+     */
+    public DescriptorProviderStateNotifier getDescriptorProviderStateNotifier();
```

Comment:
Feels like this would be better off just being part of the DescriptorProvider. Adding the method at a higher level (the configuration - which is a configuration of many things) makes the specific thing impact the more general thing, which isn't a good separation of concerns.